packages/adapter/lib/adapter/adapter.js

```diff
@@ -8731,6 +8734,53 @@ function Adapter(options) {
     this.pluginHandler = new PluginHandler(pluginSettings);
     this.pluginHandler.addPlugins(this.ioPack.common.plugins, [this.adapterDir, __dirname]); // first resolve from adapter directory, else from js-controller
 
+    /**
+     * This method returns the list of license that can be used by this adapter
+     * @param {boolean} all if return the licenses, that used by other instances (true) or only for this instance (false)
+     * @returns {object[]} list of suitable licenses
```

Comment:
it can also return undefined

Reply:
else it returns a promise with the object array

Reply:
It always returns an array. Array can be empty. Do we really want to go behind the syntax sugar?

Reply:
e. g. here it returns udnefined https://github.com/ioBroker/ioBroker.js-controller/pull/1591/files#diff-e1a840d5540c0c4c81de947b026e21a523fe711eec5e06094849c36196e27d67R8761 else it returns `Promise<object[]>`. I don't think this is syntax sugar. But if we do jsdoc we should do it correct, else it won't really support us

Reply:
Edit okay.. inside for each, fine